crypto/params.c

```diff
@@ -518,7 +535,14 @@ int OSSL_PARAM_get_uint64(const OSSL_PARAM *p, uint64_t *val)
         switch (p->data_size) {
         case sizeof(double):
             d = *(const double *)p->data;
-            if (d >= 0 && d <= INT64_MAX && d == (uint64_t)d) {
+            if (d >= 0
+                    /*
+		     * By subtracting 65535 (2^16-1) we cancel the low order
+		     * 15 bits of ULONG_MAX to avoid using imprecise floating
+		     * point values.
+                     */
+                    && d < (double)(UINT64_MAX - 65535) + 65536.
```

Comment:
I prefer 65536.0

Reply:
Okay, I'm not fussed so I'll change.  Why is there never any love for `65536e0`?

Reply:
> Okay, I'm not fussed so I'll change. Why is there never any love for `65536e0`?

Because it looks like a git reference? :grin: